hpx/runtime/applier/applier.hpp

```diff
@@ -159,7 +159,8 @@ namespace hpx { namespace applier
         }
 
         /// Schedule  threads based on the given parcel
-        void schedule_action(parcelset::parcel p, std::size_t num_thread = -1);
+        void schedule_action(parcelset::parcel p, std::size_t num_thread =
+                             (std::numeric_limits<std::size_t>::max)());
```

Comment:
> So, what are you trying to express here? Interpretation 1: num_thread should have a default value of -1. That's impossible because size_t is an unsigned type. Interpretation 

It's not impossible, `unsigned` has modulo arithmetics so each value representation `V` represents any and all values in the infinite set `2^N * k + V`.

> IIRC -1 is an integer literal

It isn't, `1` is an integer literal, `-1` is a unary expression.

> It's a hack my compiler complains about.

It's not a hack, it's an idiomatic way of generating the same value representation that this patch has, and without all the cruft.

That said, an explicit cast (of any kind) would be more explicitly and should silence any complains.